lightwood/mixers/sklearn_mixer.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np
+
+from sklearn.svm import SVC
+from sklearn.linear_model import SGDClassifier, SGDRegressor, LinearRegression
+from sklearn.naive_bayes import GaussianNB
+from sklearn.metrics import r2_score, balanced_accuracy_score
+
+from lightwood.logger import log
+from lightwood.constants.lightwood import COLUMN_DATA_TYPES
+from lightwood.mixers import BaseMixer
+
+CLASSIFICATION_MODELS = [
+    (SGDClassifier, {}),
+    (SVC, {}),
+    (GaussianNB, {}),
+]
+
+REGRESSION_MODELS = [
+    (LinearRegression, {}),
+    (SGDRegressor, {}),
+]
+
+class SklearnMixer(BaseMixer):
+    def __init__(self):
+        super().__init__()
+        self.targets = None
+        self.transformer = None
+        self.encoders = None
+
+    def fit(self, train_ds, test_ds=None):
+        """
+        :param train_ds: DataSource
+        :param test_ds: DataSource
+        """
+        self.fit_data_source(train_ds)
+
+        self.targets = {}
+
+        if test_ds is None:
+            test_ds = train_ds.subset(0.1)
```

Comment:
This should happen in `predictor.py` since multiple mixers will require a test_ds.

Reply:
Also, probably not an ideal way to do this at all, since `subset` will not "remove" the rows from the `train_ds`. And the idea with the test_ds is that it should contain different rows from train_ds.

Reply:
`DataSource.subset()`  documentation:
```
Removes :percentage: of data randomly from itself and returns a new
datasource made from that data
```

Reply:
test_ds is now required